File: chapter05-classes/instructors-examples/rational/main.py
#!/usr/bin/env python3

def gcd(a, b):
    while b != 0:
       t = b 
       b = a % b 
       a = t
    return a

def rgcd(a, b):
    if b == 0:
        return a
    else:
        return rgcd(b,a % b)


print(gcd(124,240))
print(rgcd(124,240))

File: chapter05-classes/instructors-examples/rational/main.cpp
#include<iostream>


int gcd(int a, int b)
{
    while (b != 0)
    {
       int t = b; 
       b = a % b; 
       a = t; 
   }
   return a;
}

int rgcd(int a, int b)
{
    if (b == 0)
        return a;
    else
        return rgcd(b,a % b);
}


int main()
{
    std::cout << gcd(124,240) << std::endl;
    std::cout << rgcd(124,240) << std::endl;

    return 0;
}
